core/tests/protractor_utils/ExplorationPlayerPage.js

```diff
@@ -154,83 +159,55 @@ var ExplorationPlayerPage = function() {
   };
 
   this.reportExploration = async function() {
-    await waitFor.elementToBeClickable(
-      reportExplorationButton,
-      'Report Exploration Button takes too long to be clickable');
-    await reportExplorationButton.click();
+    await action.click('Report Exploration Button', reportExplorationButton);
     let radioButton = await element.all(by.tagName('input')).get(0);
     await waitFor.visibilityOf(
       radioButton, 'Radio Buttons takes too long to appear');
-    await radioButton.click();
+    await action.click('Radio Button', radioButton);
     let textArea = element(by.tagName('textarea'));
-    await textArea.sendKeys('Reporting this exploration');
+    await action.sendKeys('Text Area', textArea, 'Reporting this exploration');
     let submitButton = await element.all(by.tagName('button')).get(1);
-    await submitButton.click();
+    await action.click('Submit Button', submitButton);
     let afterSubmitText = await element(by.tagName('p')).getText();
     expect(afterSubmitText).toMatch(
       'Your report has been forwarded to the moderators for review.');
   };
 
   this.viewHint = async function() {
-    var until = protractor.ExpectedConditions;
-    const WAIT_FOR_FIRST_HINT_MSEC = 60000;
     // We need to wait some time for the solution to activate.
-    await browser.wait(
-      until.elementToBeClickable(viewHintButton), WAIT_FOR_FIRST_HINT_MSEC,
-      '"View Hint" button takes too long to be clickable');
```

Comment:
Hi @jimbyo, I think this part is actually required for the hint funtionality to work properly. This might explain why your e2e tests are failing in the "view hint" steps.

Reply:
Ok, I'll look into it! Although its strange, I know that action.click() has a waitFors in it as well.

Reply:
Yep it does but the default timeout duration used in action.click() is shorter than what we want it to be for hints / solutions.